packages/bots/liquidator/src/services/liquidator.ts

```diff
@@ -95,53 +98,64 @@ export class Liquidator {
+        } else {
+          throw new Error(`Transaction ${sentTx} failed with status: ${receipt.status}`);
+        }
       } catch (error: any) {
         logger.error(`Liquidation failed for borrower ${params.borrower}: ${error.message}`);
         // Create a LiquidatablePool instance for the alert
         const liquidationPool: LiquidatablePool = {
           liquidations: [params],
-          comptroller: "",
+          comptroller: pool.comptroller, // Use the actual comptroller from the pool
         };
         // Send alert for the failed liquidation
         await this.alert.sendLiquidationFailure(liquidationPool, error.message);
       }
-      // logger.info(`Liquidation Succeeded for borrower ${successfulTxs}`);
-      // Inside the liquidate method after success
-      // Inside the liquidate method after success
-      // console.log("Successful Transaction", successfulTxs);
-      // Inside the liquidate method after success
-      if (successfulTxs.length > 0) {
-        // Convert the successfulTxs array to a string format
-        const msg = successfulTxs
-          .map((tx) => {
-            return (
-              `Transaction Hash: ${tx.transactionHash}\n` +
-              `Contract Address: ${tx.contractAddress}\n` +
-              `From Address: ${JSON.stringify(tx.from)}\n` + // Include the whole 'from' object
-              `To Address: ${tx.to}\n` +
-              `Status: ${tx.status}\n` +
-              `**----------------------------------**`
-            );
-          })
-          .join("\n");
-        logger.info(`Sending success alert for successful transactions: ${JSON.stringify(successfulTxs)}`);
-        // You can include the entire successfulTxs object in the alert
-        await this.alert.sendLiquidationSuccess(successfulTxs, msg);
+    }
+
+    // Process each successful transaction and send alert
+    for (const tx of successfulTxs) {
+      // Wait for the transaction receipt for each successful transaction
+      const receipt = await this.sdk.publicClient.waitForTransactionReceipt({ hash: tx.transactionHash });
+
+      // Check if the receipt status is 'success'
+      if (receipt.status === "success") {
+        // Construct the message for each successful transaction directly
+        const msg =
+          `Transaction Hash: ${tx.transactionHash}\n` +
+          `Contract Address: ${tx.contractAddress}\n` +
+          `From Address: ${JSON.stringify(tx.from)}\n` + // Include the whole 'from' object
+          `To Address: ${tx.to}\n` +
+          `Status: ${tx.status}\n` +
+          `**----------------------------------**`;
+
+        logger.info(`Sending success alert for transaction: ${tx.transactionHash}`);
+        // Send the success alert for the individual transaction
+        await this.alert.sendLiquidationSuccess([tx], msg);
+      } else {
+        logger.error(`Transaction ${tx.transactionHash} failed after receipt with status: ${receipt.status}`);
```

Comment:
Should discord alert with failure here, also printing the tx hash.

Reply:
yeah its not needed i added it for some checks will remove it